### PR TITLE
Avoid simultaneous access to shared resources by setting isFirstSession property

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
It's a bit tricky, since most of these conditions are based on static code analysis. If you see more places where this verification should be added then please let me know about it.

### CODE REVIEW

File: lib/driver.js [hunk 1]
```diff
@@ -346,7 +366,7 @@ class XCUITestDriver extends BaseDriver {
   }
 
   async startWda (sessionId, realDevice) {
-    this.wda = new WebDriverAgent(this.xcodeVersion, this.opts);
+    this.wda = new WebDriverAgent(this.xcodeVersion, Object.assign(this.opts, {isFirstSession: this.isFirstSession}));
```
Comment (imurchie) [inline lib/driver.js]: I would prefer this be done right after we figure out if it is the first session.

File: lib/driver.js [hunk 2]
```diff
@@ -799,6 +819,11 @@ class XCUITestDriver extends BaseDriver {
     this.opts.resetOnSessionStartOnly = !util.hasValue(this.opts.resetOnSessionStartOnly) || this.opts.resetOnSessionStartOnly;
     this.opts.useNewWDA = util.hasValue(this.opts.useNewWDA) ? this.opts.useNewWDA : false;
 
+    if (!this.isFirstSession && this.opts.allowTouchIdEnroll) {
```
Comment (imurchie) [inline lib/driver.js]: This too, I think, should be done earlier.

Comment (mykola-mokhnach) [inline lib/driver.js]: shall I include this argument into _this.opts_ explicitly?

File: lib/driver.js (hunk 1 above)
Comment (mykola-mokhnach) [inline lib/driver.js]: shall I include this argument into this.opts explicitly?

**Comment by mykola-mokhnach** [comment]
> It seems like, for Xcode 10+, we could also skip the build step after the first run, just doing test-without-building.

We are already trying to keep and then pick up running xcodebuild session if possible, so it shouldn't be the case

File: lib/driver.js (hunk 2 above)
Comment (mykola-mokhnach) [inline lib/driver.js]: fixed

**Comment by imurchie** [comment]
Thanks makes sense, yes.

File: lib/driver.js (hunk 1 above)
Comment (imurchie) [inline lib/driver.js]: Sure, why not?

**Comment by imurchie** [comment]
There is a test that checks that the returned options are correct (https://github.com/appium/appium-xcuitest-driver/blob/master/test/functional/basic/basic-e2e-specs.js#L45-L66). `isFirstSession` needs to be added to the "expected" caps (https://github.com/appium/appium-xcuitest-driver/blob/master/test/functional/basic/basic-e2e-specs.js#L51).

**Comment by mykola-mokhnach** [comment]
Thanks @imurchie . The test has been updated

**Pull request merged**